packages/block-editor/src/store/private-selectors.js

```diff
@@ -611,8 +613,22 @@ export function isZoomOutMode( state ) {
  *
  * @param {Object} state Editor state.
  *
- * @return {string|undefined} The section root client ID or undefined if not set.
+ * @return {string} The section root client ID.
  */
 export function getSectionRootClientId( state ) {
-	return state.settings?.[ sectionRootClientIdKey ];
+	const settingsRootClientId = state.settings?.[ sectionRootClientIdKey ];
+
+	// Specifically check that the setting was not provided to avoid
+	// cases where the provided setting is an empty string to signify
+	// the "root block" of the editor.
+	if ( settingsRootClientId !== undefined ) {
+		return settingsRootClientId;
```

Comment:
Do we explicitly provide `""` as a settingsRootClientId in the post editor today or do we leave that to the block-editor's algorithm?

Reply:
As `@wordpress/editor` provides the setting, today Site and Post Editors would use the algorithm that looks for `main`.

However, if the rendering mode is `template-locked` (like when you are editing a _Page_ in the _Site_ Editor) then the default would be `''` (empty string).

I think the question [is here](https://github.com/WordPress/gutenberg/pull/65516/files#r1768396511)

Reply:
Main main question is about "performance". In other words, it seems unnecessary to run the fallback for post and page editors in "post-only" mode (not template-locked), for template-locked, I suspect that we already have something that sets the "post-content" block as section root.

Reply:
I see what you mean. If we are in Post Editor (Posts or Pages or whatever) we are in `post-only` mode. Therefore we're going to be looking for a `main` tag which is very unlikely to exist.

Yes maybe I need to do some more digging to find where these editors end up with `''` as the sectionRootClientId.